src/client/Client.cc

```diff
@@ -3938,18 +3946,22 @@ void Client::add_update_cap(Inode *in, MetaSession *mds_session, uint64_t cap_id
 
   unsigned old_caps = cap.issued;
   cap.cap_id = cap_id;
-  cap.issued |= issued;
-  cap.implemented |= issued;
+  cap.issued = valid_issued | issued;
+  cap.implemented = valid_implemented | issued;
```

Comment:
I'd like to avoid  using valid_issued/valid_implemented. just do cap_is_valid check here. Otherwise, looks good.